### DISCUSSION TITLE
Why is the v2 documentation so bad?

### DISCUSSION THREAD

**Discussion by k-dahl** [aws/aws-sdk-go-v2#2713] [Q&A] [answered]
On https://aws.github.io/aws-sdk-go-v2/docs/ the information is extremely sparse.

Neither of the site searches work - the sidebar one does nothing when a search term is entered and enter is pressed:

![image](https://github.com/user-attachments/assets/ae57fc93-97f9-46f5-ab89-4e4acbb13b15)

The nav bar search does nothing when a search term is entered and enter is pressed:

![image](https://github.com/user-attachments/assets/91b2d280-81ab-43a8-9eb2-ce3cb8671d3a)

The API Reference link leads to this page:

https://pkg.go.dev/github.com/aws/aws-sdk-go-v2#section-readme

...which ALSO has no searchable documentation:

![image](https://github.com/user-attachments/assets/d1946503-0c01-47c7-a29a-e846aee527fe)

**Comment by stefansundin** [comment]
I was about to open a new issue or discussion about the search being broken but I found this.

It looks like the JavaScript is attempting to load https://aws.github.io/aws-sdk-go-v2/offline-search-index.9f33279f24e0567986b8fc9db212092b.json but that leads to a 404 error.

Not sure how to best get the attention of the AWS SDK team about this issue... This may not work but I am going to try to mention a random AWS person who is creating PRs in this repository:

@lucix-aws you are our only hope 😄 

This is probably an easy fix now that we know why the search isn't working.

**Comment by lucix-aws** [comment, marked as answer]
The frustration here is totally warranted. Note that we do have a strategic initiative ongoing at this time to consolidate **all** of our AWS SDK docs - so, the GitHub pages-hosted documentation we have for the v2 SDK is getting reformatted and moved to the standard AWS SDK documentation portal. That work is ongoing and should categorically solve the navigation/search issues on the page.

Unfortunately we don't have any control over the pkg.go.dev docs - the source for that is owned by Google and mirrored at https://github.com/golang/pkgsite, you can follow the instructions there if you'd like to file an issue.

To the original comment about the _current_ developer guide being informationally sparse - we'd like to hear specific feedback about what is missing, if you could share. We do also have a dedicated template for documentation if you'd like to file an [issue](https://github.com/aws/aws-sdk-go-v2/issues/new?assignees=&labels=documentation%2Cneeds-triage&projects=&template=documentation.yml&title=%28short+issue+description%29).

**Reply by stefansundin** [reply]
You do have control over https://aws.github.io/aws-sdk-go-v2/docs/ so please at least fix the search on that website. Please open that page and open your browser developer tools and you will see that https://aws.github.io/aws-sdk-go-v2/offline-search-index.9f33279f24e0567986b8fc9db212092b.json fails to load which is why the search is broken. That is probably a simple fix.

**Reply by stefansundin** [reply]
@lucix-aws Is it possible that you can address the search functionality being broken at least? That's the biggest current problem in my opinion, and it _should_ in theory be very easy to fix.

Please open https://aws.github.io/aws-sdk-go-v2/docs/ and open your browser developer tools and you will see that https://aws.github.io/aws-sdk-go-v2/offline-search-index.9f33279f24e0567986b8fc9db212092b.json fails to load which is why the search is broken.
